tests/components/whirlpool/test_climate.py

```diff
@@ -0,0 +1,236 @@
+        yield mock_aircon_api
+
+
+@pytest.fixture(name="aircon_entity")
+def aircon_entity_fixture(aircon_api):
+    """Set up whirlpool air conditioner entity."""
+    auth = Mock()
+    return AirConEntity("TEST-SAID", auth)
+
+
+async def test_async_setup_entry(hass):
+    """Test the setup of the climate entities."""
+    auth_mock = Mock()
+    auth_mock.get_said_list.return_value = ["SAID1"]
+    async_add_entities_mock = MagicMock()
+    config_mock = MagicMock(
+        data={
+            "entry_id": "0123456789",
+        }
+    )
+    hass.data[DOMAIN] = {config_mock.entry_id: {"auth": auth_mock}}
+
+    await async_setup_entry(hass, config_mock, async_add_entities_mock)
+    async_add_entities_mock.assert_called_once()
+    entities = async_add_entities_mock.call_args.args[0]
+    assert len(entities) == 1
+    assert entities[0].unique_id == "SAID1"
+
+
+async def test_async_setup_entry_no_appliances(hass):
+    """Test the setup of the climate entities when there are no appliances available."""
+    auth_mock = Mock()
+    auth_mock.get_said_list.return_value = []
+    async_add_entities_mock = MagicMock()
+    config_mock = MagicMock(
+        data={
+            "entry_id": "0123456789",
+        }
+    )
+    hass.data[DOMAIN] = {config_mock.entry_id: {"auth": auth_mock}}
+
+    await async_setup_entry(hass, config_mock, async_add_entities_mock)
+    async_add_entities_mock.assert_not_called()
+
+
+async def test_name(aircon_entity, aircon_api):
+    """Test name property."""
+    aircon_api.return_value.connect = AsyncMock()
+    aircon_api.return_value.fetch_name = AsyncMock(return_value="TestZone")
+
+    assert aircon_entity.name == "TEST-SAID"
+    await aircon_entity.async_added_to_hass()
+    assert aircon_entity.name == "TestZone"
+
+
+async def test_unique_id(aircon_entity, aircon_api):
+    """Test name property."""
+    aircon_api.return_value.said = "TEST-SAID-123"
+    assert aircon_entity.unique_id == "TEST-SAID-123"
+
+
+async def test_static_attributes(aircon_entity: AirConEntity, aircon_api):
+    """Test getting attributes."""
+    assert (
+        aircon_entity.supported_features
+        == SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE | SUPPORT_SWING_MODE
+    )
+    assert aircon_entity.hvac_modes == [
+        HVAC_MODE_COOL,
+        HVAC_MODE_HEAT,
+        HVAC_MODE_FAN_ONLY,
+        HVAC_MODE_OFF,
+    ]
+    assert aircon_entity.fan_modes == [FAN_AUTO, FAN_HIGH, FAN_MEDIUM, FAN_LOW, FAN_OFF]
+    assert aircon_entity.swing_modes == [SWING_HORIZONTAL, SWING_OFF]
+    assert aircon_entity.temperature_unit == TEMP_CELSIUS
+    assert aircon_entity.target_temperature_step == 1
+    assert aircon_entity.target_humidity_step == 10
+    assert aircon_entity.min_temp == 16
+    assert aircon_entity.max_temp == 30
+
+
+async def test_get_attributes(aircon_entity: AirConEntity, aircon_api):
+    """Test getting attributes."""
+    aircon_api.return_value.get_online.return_value = False
```

Comment:
Setup the integration and check the values in the state machine using `hass.states.get`

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Reply:
done